tests/nfs/test_earmark_feature.py

```diff
@@ -0,0 +1,148 @@
+    nfs_name = "cephfs-nfs"
+    nfs_export = "/export"
+    nfs_server_name = nfs_node.hostname
+    subvolume_group = "ganeshagroup"
+    subvolume_name = "subvolume"
+    earmark = config.get("earmark")
+
+    fs_util.create_subvolumegroup(
+        client=clients[0], vol_name=fs_name, group_name=subvolume_group
+    )
+    fs_util.create_subvolume(
+        client=clients[0],
+        vol_name=fs_name,
+        subvol_name=subvolume_name,
+        validate=True,
+        group_name=subvolume_group,
+    )
+    fs_util.set_subvolume_earmark(
+        client=clients[0],
+        vol_name=fs_name,
+        subvol_name=subvolume_name,
+        group_name=subvolume_group,
+        earmark=earmark,
+    )
+
+    subvolume_earmark = fs_util.get_subvolume_earmark(
+        client=clients[0],
+        vol_name=fs_name,
+        subvol_name=subvolume_name,
+        group_name=subvolume_group,
+    )
+
+    if operation == "verify_earmark":
+        if earmark not in subvolume_earmark:
+            log.error(f'earmark "{earmark}" not found on subvolume {subvolume_name}')
+            # raise OperationFailedError(f"earmark \"{earmark}\" not found on subvolume {subvolume_name}")
+            return 1
+
+        log.info(f'earmark "{earmark}" found on subvolume {subvolume_name}')
+        return 0
+
+    if operation == "rename_earmark":
+        earmark2 = "nfs"
+        fs_util.remove_subvolume_earmark(
+            client=clients[0],
+            vol_name=fs_name,
+            subvol_name=subvolume_name,
+            group_name=subvolume_group,
+        )
+
+        fs_util.set_subvolume_earmark(
+            client=clients[0],
+            vol_name=fs_name,
+            subvol_name=subvolume_name,
+            group_name=subvolume_group,
+            earmark=earmark2,
+        )
+
+    try:
+        # Setup nfs cluster
+        Ceph(clients[0]).nfs.cluster.create(
+            name=nfs_name, nfs_server=nfs_server_name, ha=False, vip=None
+        )
+        sleep(3)
+
+        if operation == "override_earmark":
+            earmark2 = "smb"
+            fs_util.set_subvolume_earmark(
+                client=clients[0],
+                vol_name=fs_name,
+                subvol_name=subvolume_name,
+                group_name=subvolume_group,
+                earmark=earmark2,
+            )
+
+        log.info(f"subvolume earmark is {subvolume_earmark}")
+
+        sub_volume_path = fs_util.get_subvolume_info(
+            client=clients[0],
+            vol_name=fs_name,
+            subvol_name=subvolume_name,
+            group_name=subvolume_group,
+        ).get("path")
+
+        try:
+            create_nfs_export(
+                clients[0], fs_name, nfs_name, nfs_export, sub_volume_path, ""
+            )
+            log.info(
+                f"nfs export {nfs_export} has been created for subvolume path {nfs_export}"
+            )
+
+            Ceph(clients[0]).nfs.export.delete(nfs_name, nfs_export)
```

Comment:
Can export delete part be moved to "finally" block. 

Eg - 
    finally:
        log.info("Cleaning up")
        sleep(3)
        cleanup_cluster(clients, nfs_mount, nfs_name, nfs_export)
        log.info("Cleaning up successfull")